src/App.tsx

```diff
@@ -1,24 +1,78 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+class App extends React.Component {
+  state = {
+    timerId: undefined,
+    currentTime: new Date().toLocaleTimeString(),
+    isClockVisible: true,
+  };
+
+  componentDidMount() {
+    this.startClockTimer();
+  }
+
+  componentDidUpdate() {
+    if (this.state.isClockVisible && !this.state.timerId) {
+      this.startClockTimer();
+    }
+  }
+
+  startClockTimer = () => {
+    const id = setInterval(() => {
+      const time = new Date().toLocaleTimeString();
+
+      // eslint-disable-next-line
+      console.log(time);
+
+      this.setState({
+        timerId: id,
+        currentTime: time,
+      });
+    }, 1000);
+  };
```

Comment:
this seems like the logic of your clock. but it is in the App for some reason.
think about components as separate pieces. I want to take this clock and use it somewhere else. but in your code, some parts of it are in a different component